urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/examples/SepStatistics.java

```diff
@@ -0,0 +1,190 @@
+        following occurs: unique distribution is found (no non-trivial
+        equidistributions) or until the same number of equidistribution classes
+        is found twice in a row or until maxLengthN is hit. Note that if the
+        same number of classes is found twice in a row and there is a
+        non-trivial equidistribution, this class should be checked to further
+        lengths.
+    */
+    
+    public static void main(String[] args) {
+        int k = 6;
+        int maxLengthN = 11;
+        
+        ArrayList<PermutationClass> classesToCheck = new ArrayList<>();
+        
+        classesToCheck.add(new PermutationClass(new Permutation("2413"), new Permutation("3142")));
+        classesToCheck.add(new PermutationClass(new Permutation("1324"), new Permutation("4231")));
+        classesToCheck.add(new PermutationClass(new Permutation("1432"), new Permutation("2341"), new Permutation("4123"), new Permutation("3214")));
+        
+        for (Permutation p : new Permutations(3,5)) {
+            if (PermUtilities.isSymmetryRep(p)) {
+                classesToCheck.add(new PermutationClass(p));
+            }
+        }
+        
+        for (PermutationClass C : classesToCheck) {
+            System.out.println("\nChecking class: " + C);
+            int n = k+1;
+            HashMap distClasses = findDistClasses(k, n, C);
+            int last = 0;
+            int numDistinct = distClasses.keySet().size();
+            System.out.println("\t\t" + numDistinct + " distinct classes of length " + n + ".");
+            while (last != numDistinct && n < maxLengthN && hasNontrivialDistEq(distClasses)) {    
+                n++;
+                last = numDistinct;
+                distClasses = findDistClasses(k, n, C);
+                numDistinct = distClasses.keySet().size();
+                System.out.println("\t\t" + numDistinct + " distinct classes of length " + n + ".");
+            }
+            if (hasNontrivialDistEq(distClasses)) {
+                if (last != numDistinct) {
+                    System.out.println("\t <!> maxN not high enough. <!>");
+                }
+                System.out.println("\t NOT Uniquely Distributed");
+                printClasses(distClasses, false, "\t\t");
+            } else {
+                System.out.println("\t Uniquely Distributed");
+            }   
+        }
+    }
+    
+    public static boolean hasNontrivialDistEq(HashMap<OccurrenceCounts, HashSet<Permutation>> distClasses) {
+        for(OccurrenceCounts cs : distClasses.keySet()) {
+            HashSet<Permutation> symReps = new HashSet<>();
+            HashSet<Permutation> permsToPrint = new HashSet<>();
+            for(Permutation p : distClasses.get(cs)) {
+                Permutation r = PermUtilities.symmetryRep(p);
+                if (!symReps.contains(r)){
+                    permsToPrint.add(p);
+                    symReps.add(r);
+                }
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 80:86, 141:147.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.